internal/generator/generator.go

```diff
@@ -60,21 +61,34 @@ func (g *Generator) Generate(examples ...*linker.LinkedExample) []*Suite {
 		var deps = Dependencies([]Dependency{Dependency(g.conf.BasePkg)})
 		deps = append(deps, normalizeDeps(moduleName, e.Dependencies())...)
 
-		result = append(result, &Suite{
+		var setupDeps = Dependencies([]Dependency{Dependency(g.conf.BasePkg)})
```

Comment:
Basically, to simplify this we can add into the example some function component `IsSubSuite` that will return true if one of the parents includes this example and false for other cases.

So my point to have here a clean section with branches for `tests`, `suites`, and `subsuits`.

Reply:
subSuite and suite has the only difference in containing parent suite. I just added more comments to the code.